### ISSUE TITLE
COAM \ Earned Credits - Internal User can review EC request

### ISSUE THREAD

**Issue by NicoleGovvy** [bcgov/cas-coam#63]
#### Description:

CAS staff can review details of an EC issuance request before making recommendations and a final decision.

#### Acceptance Criteria:

As a CAS staff member, given that an operator has requested an EC issuance, when they click 'Review EC request' on the compliance obligation grid, then they can view details of the request submitted by the operator on a new page.

- [ ] 'Review Earned Credit request' page displays as a vertical tab, below 'Review [year] Compliance Summary
- [ ] A button called 'View annual emissions report' can be clicked to view the report [opens as a pop-up]
- [ ] Earned Credits heading

Underneath this heading the following is displayed:
- [ ] Number of earned credits requested
- [ ] Date requested
- [ ] Status of issuance
- [ ] BCCR Trading name
- [ ] BCCR holding account number

- [ ] A new heading called 'CAS Analyst'
- [ ] CAS analyst recommendation: radio buttons showing: 
`- [ ] `dd

#### Development Checklist:

- [ ] Checklist item
- [ ] Checklist item
- [ ] Checklist item
- [ ] Meets the DOD

#### Definition of Ready (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

#### Definition of Done (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Blocked By/Blocking
Blocked by #62 

#### Notes:

-

**Comment by NicoleGovvy** [comment]
When the 'decline' button pressed by the director, the request will show up in the operator's Track Status of Issuance as 'declined' (use appropriate status wording), and they will see the information about how to get more info about the decline (email ghgregulator) and how to appeal.

When the 'approve' button is pressed, the request will show up as approved waiting credit issuance to the operator.

When the approve button is pressed, credits will either be automatically issued in BCCR using the API details (ideal scenario), OR a notification will be sent to CAS staff to manually issue the credits in BCCR

**Comment by NicoleGovvy** [comment]
@NicoleGovvy the above comment needs a new ticket.

**Comment by patrickisaac** [comment]
Refinement Notes:
- Needing dev checklist

**Comment by fviduya** [comment]
I have tested this in the DEV environment, [here is the test document for reference](https://buttoninc-my.sharepoint.com/:w:/g/personal/francesca_button_is/EWP-WhbTeEFAsmwCx23qCzsBRaSoN-J8Oo51RgzqdlSHaw?e=VKbMMd&nav=eyJoIjoiMTQwNDk1MTcyOSJ9). 

I can verify that:
- [x] 'Review Earned Credit request' page displays as a vertical tab, below 'Review [year] Compliance Summary
- [x] A button called 'View annual emissions report' can be clicked to view the report [opens as a pop-up]
- [x] Earned Credits subheading
- [x] Underneath the Earned Credits subheading, the following is displayed:
   - [x] Earned Credits
   - [x] Status of issuance
   - [x] BCCR Trading name
   - [x] BCCR holding account number
- [x] A new subheading 'Review by Analyst' is visible, and the following is displayed below:
   - [x] Field - CAS analyst recommendation: radio buttons showing: 
      - [x] Ready to approve
      - [x] Require changes/corrections
   - [x] Field - CAS Analyst's comments: a free-text box to enter comments, and is accessible by CAS Analyst only
   - [x] `Continue` button is visible for the analyst to save data
   - [ ] Once the analyst clicks `Continue`, their IDIR and date/time are saved - **The `Continue` button does not save data**
- [x] A new subheading `Review by Director' 
   - [x] Field - CAS Director comments: a free-text box for the director to enter comments, and is accessible by CAS Director only.
   - [x] `Approve` and `Decline` buttons are visible and accessible by CAS Director role only.
- [ ] The link name is changed from `Request Issuance of Credits` to `Review Credits Issuance Request` for internal user - **Created #217 for this**

Noticed the following during testing:
- The `Continue` button does not save the data entered by the analyst. Wanted to check with @nataliepiguliak if that's part of the incoming ticket which is #65
- The breadcrumbs should display the operation name at the end, per the wireframe, instead shows "Request Issuance Review Summary"
- Missing operation name in the top left of the page itself
- When user navigates to the Review Credits Issuance Request page, the Compliance Summary year changes to 2023 from 2024.

Hi @vesselak, do we want to create a separate ticket to tackle all of the mentioned issues above? Moving to Pending PO for review of test. Thank you!

**Comment by nataliepiguliak** [comment]
@fviduya regarding your question on the `Continue` button, it will be implemented at part of #166

**Comment by vesselak** [comment]
@fviduya would you please create a ticket for:

- The breadcrumbs should display the operation name at the end, per the wireframe, instead shows "Request Issuance Review Summary"
- Missing operation name in the top left of the page itself
- When user navigates to the Review Credits Issuance Request page, the Compliance Summary year changes to 2023 from 2024.

**Comment by fviduya** [comment]
Hi @vesselak, I've created #217 for the above mentioned tasks.

**Comment by vesselak** [comment]
Reviewed the ticket with QA and updated the AC to align with the approved design in Figma. 

Accepting ticket with the fixes documented in #166 & #217 & #218
